frontend/ic_vetkeys/src/utils/utils.ts

```diff
@@ -306,6 +306,34 @@ export function augmentedHashToG1(
     return pt;
 }
 
+/**
+ * Verify a BLS signature
+ *
+ * A VetKey is in the end a valid BLS signature; this function checks that a
+ * provided BLS signature is the valid one for the provided public key and
+ * message.
+ *
+ * When a VetKey struct is created (VetKey.decryptAndVerify) the signature (aka
+ * vetkey) is already verified, so using this function is only necessary when
+ * using a vetkey as a VRF or for threshold BLS signatures.
+ */
+export function verifyBlsSignature(
```

Comment:
Should we add dedicated tests for this new public function, or are the indirect tests we already have sufficient?

Reply:
IMO the existing usage in decryptAndVerify was sufficient for how it was, but did not cover the case for where we passed an array of bytes instead of a point. I added a test.